Description: use glew to init
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2020-04-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/cg_x11.c
+++ b/src/cg_x11.c
@@ -1,5 +1,6 @@
 #define GL_GLEXT_PROTOTYPES
 #define GLX_GLXEXT_PROTOTYPES
+#include <GL/glew.h>
 #include "cg_x11.h"
 #include "cg_constants.h"
 #include <stdlib.h>
@@ -13,7 +14,6 @@
 int cg_screen_mode;
 GLXContext cg_context;
 
-
 void SetScreenSize(int width, int height, XVisualInfo *visual){
 	XRRScreenSize *sizes;
 	int snum, rnum, i;
@@ -132,6 +132,8 @@
 	cg_window            = XCreateWindow(cg_display, RootWindow(cg_display, visual->screen),
 	                                     0, 0, width, height, 0, visual->depth, InputOutput,
 	                                     visual->visual, CWBorderPixel | CWColormap | CWEventMask, &swa);
+	PFNGLXCREATECONTEXTATTRIBSARBPROC glXCreateContextAttribsARB = 0;
+	glXCreateContextAttribsARB = (PFNGLXCREATECONTEXTATTRIBSARBPROC) glXGetProcAddressARB( (const GLubyte *) "glXCreateContextAttribsARB" );
 	if(!cg_window) {
 		return(-1);
 	}
--- a/src/livin.c
+++ b/src/livin.c
@@ -1,4 +1,5 @@
 #define GL_GLEXT_PROTOTYPES
+#include <GL/glew.h>
 #include <GL/gl.h>
 #include <math.h>
 #include <stdio.h>
@@ -91,6 +92,14 @@
 	char greetingspath[21];
 
 	/* OpenGL related initialization */
+	glewExperimental = GL_TRUE;
+    GLenum err = glewInit();
+    if(GLEW_OK != err)
+    {
+        //Problem: glewInit failed, something is seriously wrong.
+        printf("Error: %s\n", glewGetErrorString(err));
+    }
+    printf( "Status: Using GLEW %s\n", glewGetString(GLEW_VERSION));
 	glClearColor(0.0, 0.0, 0.0, 0.0);
 	glEnable(GL_DEPTH_TEST);
 	glEnable(GL_BLEND);
--- a/src/explosion.c
+++ b/src/explosion.c
@@ -1,4 +1,5 @@
 #define GL_GLEXT_PROTOTYPES
+#include <GL/glew.h>
 #include <GL/gl.h>
 #include <stdlib.h>
 #include "shader.h"
--- a/src/squares.c
+++ b/src/squares.c
@@ -1,4 +1,5 @@
 #define GL_GLEXT_PROTOTYPES
+#include <GL/glew.h>
 #include <GL/gl.h>
 #include "squares.h"
 
